protocol-tokio/examples/client.rs

```diff
@@ -30,7 +30,10 @@ fn main() {
             Connection::connect(stream)
         })
         .and_then(|connection| {
-            let (sink, stream) = connection.split();
+            let (sink, stream): (
+                OutboundSink<_, u8, u8, u8, u8>,
```

Comment:
This is not very illustrative as an example. Maybe mockchain types should help here. It will also serve to verify soundness of the type work.